server/post.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/robfig/cron"
+)
+
+func (p *Plugin) startCron() error {
+	c := cron.New()
+	if err := c.AddFunc("@weekly", func() { // Run once a week, midnight between Sat/Sun
+		if err := p.sendAnalytics(); err != nil {
+			p.API.LogError("can't send post", "err", err.Error())
+		}
+		p.newSession()
+	}); err != nil {
+		return err
+	}
+	c.Start()
+	return nil
+}
+
+func (p *Plugin) buildAnalyticMsg() (string, *model.AppError) {
+	data, err := p.prepareData()
+	if err != nil {
+		return "", err
+	}
+
+	p.currentAnalytic.RLock()
+	m := fmt.Sprintf("## Analytics since %s at %s\n", p.currentAnalytic.Start.Format("2 January"), p.currentAnalytic.Start.Format("15:04"))
```

Comment:
Presumably out of scope here, but what do you think about using https://docs.mattermost.com/developer/message-attachments.html for formatting the message?

Reply:
Thanks, i have not found this doc, I will use tabulation representation! https://github.com/manland/mattermost-plugin-analytics/issues/22